mesonbuild/modules/python.py

```diff
@@ -513,10 +513,13 @@ def find_installation(self, interpreter, state, args, kwargs):
             python = ExternalProgram(name_or_path, silent = True)
 
             if not python.found() and mesonlib.is_windows():
-                pythonpath = self._get_win_pythonpath(name_or_path)
-                if pythonpath is not None:
-                    name_or_path = pythonpath
-                    python = ExternalProgram(name_or_path, silent = True)
+                try:
+                    pythonpath = self._get_win_pythonpath(name_or_path)
+                    if pythonpath is not None:
+                        name_or_path = pythonpath
+                        python = ExternalProgram(name_or_path, silent = True)
+                except FileNotFoundError:
```

Comment:
I guess FileNotFoundError is raised by Popen_safe in _get_win_pythonpath()? You should handle the error there instead imo.